### ISSUE TITLE
How do you create a collection idempotently?

### ISSUE THREAD

**Issue by Qix-** [arangodb/arangojs#735]
Getting a collection that doesn't exist will fail, but it also seems that `db.createCollection()` called on an existing collection will also fail.

In the case of a total datacenter recovery, or in a new datacenter initialization, or a migration, or whatever, there will be many services attempting to self-heal and bring the collections up to date, which might include creating (new) collections.

In Mongo, there was no difference between a "created" and "missing" collection. If you inserted a document into a "missing" collection, the collection was created automatically. This made writing fault-tolerant services trivial, at least in that regard (Mongo has a slew of other problems though).

However, in Arango, this is seemingly difficult/impossible to do correctly. I understand that there are different types of collections so automatic creation isn't really an option, but how are you supposed to go about doing this?

Creating collections manually isn't an option.

**Comment by Qix-** [comment]
This actually seems to be a much larger design issue in Arango...

In terms of permissions, there is only the granularity of "administer" and "access". Self-healing via collection creation isn't possible without also allowing the connection to delete collections or do other harmful/security violating things.

Whoof. Arango might be a mistake at this point. Please consider adding more permissions granularity to Arango, it's not secure as-is...

**Comment by Qix-** [comment]
Yeah, not being able to specify "create collection but don't allow this user to drop collections or grant privileges" is a huge security hole in this case. I can't write a fault-tolerant/self-healing service that isn't also able to cause destructive changes with Arango. This is entirely possible with just about every other DBMS I know of.

Please, _please_ consider increasing the granularity of the security controls of Arango (I know this isn't related to the JS library in particular). Arango seems perfect for our use-case but I can't write robust software with it :/

**Comment by pluma** [comment]
Sorry, I don't see anything actionable here from the arangojs side of things so I'm moving this to the main arangodb repository. If you have questions about specific deployment scenarios, please also check out the [ArangoDB Community Slack](https://slack.arangodb.com).
